scene/main/viewport.cpp

```diff
@@ -1104,6 +1105,12 @@ String Viewport::_gui_get_tooltip(Control *p_control, const Vector2 &p_pos, Cont
 	while (p_control) {
 		tooltip = p_control->get_tooltip(pos);
 
+		//Temporary solution for PopupMenus
+		if (this->get_class_name() == "PopupMenu") {
```

Comment:
Why bother with this? Can't you just do

```cpp
if (PopupMenu *menu = Object::cast_to<PopupMenu>(this)){
	tooltip = menu->get_tooltip(pos);
}
```

Reply:
Note that we prefer to avoid assignments in `if` conditionals, so this hould be:
```cpp
PopupMenu *menu = Object::cast_to<PopupMenu>(this);
if (menu) {
	menu->get_tooltip(pos);
}
```

Reply:
You're right. I may have such mistakes as I am new to the project. Thank you info.